src/library/scala/collection/mutable/HashMap.scala

```diff
@@ -72,6 +72,17 @@ extends AbstractMap[A, B]
     else Some(e.value)
   }
 
+  override def getOrElseUpdate(key: A, defaultValue: => B): B = {
```

Comment:
In case the element was missing from the map, the hash code (with some added variance) was calculated first for verification, and again for the `put` operation.